src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactoryHelpers.cs

```diff
@@ -0,0 +1,345 @@
+    /// </summary>
+    internal static class BindingFactoryHelpers
+    {
+         // Bind a trigger argument to various parameter types. 
+        // Handles either T or T[], 
+        internal static ITriggerDataArgumentBinding<TTriggerValue> GetTriggerArgumentBinding<TMessage, TTriggerValue>(
+            ITriggerBindingStrategy<TMessage, TTriggerValue> bindingStrategy,
+            ParameterInfo parameter,
+            IConverterManager converterManager,
+            out bool singleDispatch)
+        {
+            ITriggerDataArgumentBinding<TTriggerValue> argumentBinding = null;
+            if (parameter.ParameterType.IsArray)
+            {
+                // dispatch the entire batch in a single call. 
+                singleDispatch = false;
+
+                var elementType = parameter.ParameterType.GetElementType();
+
+                var innerArgumentBinding = GetTriggerArgumentElementBinding<TMessage, TTriggerValue>(elementType, bindingStrategy, converterManager);
+
+                argumentBinding = new ArrayTriggerArgumentBinding<TMessage, TTriggerValue>(bindingStrategy, innerArgumentBinding, converterManager);
+
+                return argumentBinding;
+            }
+            else
+            {
+                // Dispatch each item one at a time
+                singleDispatch = true;
+
+                var elementType = parameter.ParameterType;
+                argumentBinding = GetTriggerArgumentElementBinding<TMessage, TTriggerValue>(elementType, bindingStrategy, converterManager);
+                return argumentBinding;
+            }
+        }
+
+        // Bind a T. 
+        private static SimpleTriggerArgumentBinding<TMessage, TTriggerValue> GetTriggerArgumentElementBinding<TMessage, TTriggerValue>(
+            Type elementType,
+            ITriggerBindingStrategy<TMessage, TTriggerValue> bindingStrategy,
+            IConverterManager converterManager)
+        {
+            if (elementType == typeof(TMessage))
+            {
+                return new SimpleTriggerArgumentBinding<TMessage, TTriggerValue>(bindingStrategy, converterManager);
+            }
+            if (elementType == typeof(string))
+            {
+                return new StringTriggerArgumentBinding<TMessage, TTriggerValue>(bindingStrategy, converterManager);
+            }
+            else
+            {
+                // Default, assume a Poco
+                return new PocoTriggerArgumentBinding<TMessage, TTriggerValue>(bindingStrategy, converterManager, elementType);
+            }
+        }
+
+        // Create a binding rule to an IAsyncCollector<T>, where the user parameter's type is resolved to a T via the ConverterManager.
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "converterManager")]
+        internal static IBinding BindCollector2<TAttribute, TMessage>(
```

Comment:
Why is this still named with a 2 suffix? I thought you were getting rid of the old types?